aws/resource_aws_glue_dev_endpoint_test.go

```diff
@@ -0,0 +1,968 @@
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueDevEndpointConfig_ExtraJarsS3Path(rName, extraJarsS3Path),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+					resource.TestCheckResourceAttr(resourceName, "extra_jars_s3_path", extraJarsS3Path),
+				),
+			},
+			{
+				Config: testAccGlueDevEndpointConfig_ExtraJarsS3Path(rName, extraJarsS3PathUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+					resource.TestCheckResourceAttr(resourceName, "extra_jars_s3_path", extraJarsS3PathUpdated),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccGlueDevEndpoint_ExtraPythonLibsS3Path(t *testing.T) {
+	var endpoint glue.DevEndpoint
+
+	rName := acctest.RandomWithPrefix(GlueDevEndpointResourcePrefix)
+	extraPythonLibsS3Path := "foo"
+	extraPythonLibsS3PathUpdated := "bar"
+	resourceName := "aws_glue_dev_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueDevEndpointConfig_ExtraPythonLibsS3Path(rName, extraPythonLibsS3Path),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+					resource.TestCheckResourceAttr(resourceName, "extra_python_libs_s3_path", extraPythonLibsS3Path),
+				),
+			},
+			{
+				Config: testAccGlueDevEndpointConfig_ExtraPythonLibsS3Path(rName, extraPythonLibsS3PathUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+					resource.TestCheckResourceAttr(resourceName, "extra_python_libs_s3_path", extraPythonLibsS3PathUpdated),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccGlueDevEndpoint_GlueVersion(t *testing.T) {
+	var endpoint glue.DevEndpoint
+
+	rName := acctest.RandomWithPrefix(GlueDevEndpointResourcePrefix)
+	resourceName := "aws_glue_dev_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccGlueDevEndpointConfig_GlueVersion(rName, "1"),
+				ExpectError: regexp.MustCompile(`attribute glue_version must match version pattern X.X`),
+			},
+			{
+				Config: testAccGlueDevEndpointConfig_GlueVersion(rName, "1.0"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+
```

Comment:
extraneous space

Reply:
Sorry, I hope I caught now all the spaces. Would probably be good to have a linter for this at some point similar to: https://github.com/bombsimon/wsl